application/core/services/novel_service.py

```
@@ -370,7 +370,10 @@ def update_novel_stage(self, novel_id: str, stage: str) -> NovelDTO:
         if novel is None:
             raise EntityNotFoundError("Novel", novel_id)
 
-        novel.stage = NovelStage(stage)
+        try:
+            novel.stage = NovelStage(stage)
+        except ValueError:
+            raise ValueError(f"Invalid stage value: {stage}. Valid values: {[s.value for s in NovelStage]}")
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)